apps/dashboard/lib/trpc/routers/workspace/onboarding.ts

```diff
@@ -0,0 +1,93 @@
+import { createKeyInputSchema } from "@/app/(app)/apis/[apiId]/_components/create-key/create-key.schema";
+import { db } from "@/lib/db";
+import { TRPCError } from "@trpc/server";
+import { z } from "zod";
+import { requireUser, t } from "../../trpc";
+import { createApiCore } from "../api/create";
+import { createKeyCore } from "../key/create";
+
+const createWorkspaceWithApiAndKeyInputSchema = z.object({
+  workspaceId: z.string(),
+  apiName: z
+    .string()
+    .min(3, "API name must be at least 3 characters")
+    .max(50, "API name must not exceed 50 characters"),
+  ...createKeyInputSchema.omit({ keyAuthId: true }).shape,
+});
+
+export const onboardingKeyCreation = t.procedure
+  .use(requireUser)
+  .input(createWorkspaceWithApiAndKeyInputSchema)
+  .mutation(async ({ input, ctx }) => {
+    const { workspaceId, apiName, ...keyInput } = input;
+
+    try {
+      return await db.transaction(async (tx) => {
+        // Validate that the workspace exists and user has access to it
+        const workspace = await tx.query.workspaces
+          .findFirst({
+            where: (table, { and, eq, isNull }) =>
+              and(
+                eq(table.id, workspaceId),
+                eq(table.orgId, ctx.tenant.id),
+                isNull(table.deletedAtM),
+              ),
+          })
+          .catch((_err) => {
+            throw new TRPCError({
+              code: "INTERNAL_SERVER_ERROR",
+              message:
+                "Failed to validate workspace access. If this issue persists, please contact support@unkey.dev",
+            });
+          });
+
+        if (!workspace) {
+          throw new TRPCError({
+            code: "NOT_FOUND",
+            message: "Workspace not found or you don't have access to it",
+          });
+        }
+
+        // Create workspace context for API and key creation
+        const workspaceCtx = {
+          ...ctx,
+          workspace: { id: workspaceId },
+        };
+
+        // Create API
+        const apiResult = await createApiCore({ name: apiName }, workspaceCtx, tx);
+
+        // Create key using the keyAuthId from the API
+        const keyAuth = {
+          id: apiResult.keyAuthId,
+          storeEncryptedKeys: false, // Default for new APIs. Can be activated by unkey with a support ticket.
+        };
+
+        const keyResult = await createKeyCore(
+          {
+            ...keyInput,
```

Comment:
this api smells.. why do we need to enrich he first arg wit the keyAuthId if we're passing in the entire keyAuth as last arg?

Reply:
You are right. `keyAuthId: apiResult.keyAuthId` this part not even being used lemme get rid of that.